applications/wide_baseline_visual_odometry_test.cpp

```diff
@@ -0,0 +1,130 @@
+ *     the following disclaimer in the documentation and/or other materials provided with the distribution.
+ * 
+ *  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or
+ *     promote products derived from this software without specific prior written permission.
+ * 
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, *  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ *  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ *  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  Authors:
+ *
+ *  Bruno Silva
+ *  Marcos Henrique
+ */
+
+#include <cstdio>
+#include <cstdlib>
+#include <Eigen/Geometry>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <geometry.h>
+#include <config_loader.h>
+#include <rgbd_loader.h>
+#include <event_logger.h>
+#include <wide_baseline_tracker.h>
+#include <wide_baseline_visual_odometry.h>
+#include <reconstruction_visualizer.h>
+
+using namespace std;
+using namespace cv;
+
+/**
+ * This program shows the use of camera pose estimation using optical flow visual odometry.
+ * @param .yml config. file (from which index_file is used)
+ */
+int main(int argc, char **argv)
+{
+	EventLogger& logger = EventLogger::getInstance();
+	logger.setVerbosityLevel(pcl::console::L_DEBUG);
+	
+	RGBDLoader loader;
+	Intrinsics intr(0);
+	WideBaselineVisualOdometry vo(intr);
+	ReconstructionVisualizer visualizer;
+	string index_file;
+	Mat frame, depth;
+	int log_stats;
+	Ptr<cv::FeatureDetector> feature_detector;
+	Ptr<cv::DescriptorExtractor> descriptor_extractor;
+	Ptr<cv::DescriptorMatcher> descriptor_matcher;
+
+
+	if(argc != 2)
+	{
+		logger.print(pcl::console::L_INFO, "[wide_baseline_visual_odometry_test.cpp] Usage: %s <path/to/config_file.yaml>\n", argv[0]);
+		exit(0);
+	}
+	ConfigLoader param_loader(argv[1]);
+	param_loader.checkAndGetString("index_file",index_file);
+	param_loader.checkAndGetInt("log_stats",log_stats);
+	param_loader.checkAndGetFeatureDetector("feature_detector",feature_detector);
+	param_loader.checkAndGetDescriptorExtractor("descriptor_extractor",descriptor_extractor);
+	param_loader.checkAndGetDescriptorMatcher("descriptor_matcher",descriptor_matcher);
+	
+	WideBaselineTracker wide_baseline_tracker(feature_detector, descriptor_extractor, descriptor_matcher, log_stats);
```

Comment:
WideBaselineVisualOdometry should be composed of WideBaselineTracker
(what happens if the programmer doesn't call setTracker?)

Reply:
The _WideBaselineVisualOdometry_ class  was implemented with a constructor that has a _WideBaselineTracker_ as an attribute. Therefore, the programmer can set a tracker without the _setTracker_ method.
It is also possible, instantiate an object without the _WideBaselineTracker_ parameter and not call the _setTracker_ method. In this case, we have the default _WideBaselineTracker_ with a ORB feature detector and descriptor extractor and a BRUTEFORCE matcher.

Reply:
The relation implemented between the _WideBaselineVisualOdometry_ and _WideBaselineTracker_ classes was aggregation. 
Do I have to change the relation between these classes for composition?

Reply:
Since there is the option to use a default tracker, it is not necesary.